docs/modules/clusters/pages/ucn-static-programmatic.adoc

```diff
@@ -0,0 +1,70 @@
+= Programmatic Configuration
+:description: You can configure your {ucn} and data structures programmatically. You can specify JAR, JARs In ZIP, and Class resources in programmatic configuration.
+:page-enterprise: true
+:page-beta: true
+
+{description}
+
+You must create a _member.java_ file containing the configuration and save it to your class configuration folder. 
+
+The file will be similar to the following:
+
+[source,java]
+----
+package com.hazelcast.namespace.staticconfig.classconfig;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.config.UserCodeNamespaceConfig;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+
+public class Member {
+
+    public static void main(String[] args) {
+        Config config = new Config();
+        config.getNamespacesConfig().setEnabled(true); <1>
+
+        // construct a namespace configuration
+        UserCodeNamespaceConfig userCodeNamespaceConfig = new UserCodeNamespaceConfig();
+        userCodeNamespaceConfig.setName("namespace_name"); <2>
+        userCodeNamespaceConfig.addClass(loadClass("usercodenamespaces.IncrementingEntryProcessor"));
+        config.getNamespacesConfig().addNamespaceConfig(userCodeNamespaceConfig); <3>
+
+        // referencing the new UserCodeNamespaceConfig in a map config
+        config.getMapConfig("map_name").setUserCodeNamespace("namespace_name"); <4>
+
+        HazelcastInstance member = Hazelcast.newHazelcastInstance(config);
+        // putting key/value in the map that client can execute on
+        member.getMap("map_name").put("key", 0); <5>
+    }
+
+    /**
+     * Helper code to load the class from outside the class path.
+     * @return the Class
+     */
+    private static Class loadClass(String className) {
+        String basePath = System.getProperty("user.dir");
+        String directoryPath = basePath + "/src/main/class/";
+
+        File directory = new File(directoryPath);
+        try {
+            URL url = directory.toURI().toURL();
+            ClassLoader parentClassLoader = Member.class.getClassLoader();
+            URLClassLoader classLoader = new URLClassLoader(new URL[]{url}, parentClassLoader);
+            return classLoader.loadClass(className);
+        } catch (MalformedURLException | ClassNotFoundException e) {
+            throw new RuntimeException("Error loading class", e);
+        }
+    }
+}
+----
+<1> Enables {ucn} on `Config`
+<2> Creates a `UserCodeNamespaceConfig` with a name of `namespace_name` and adds the external JAR URL containing the resources
+<3> Adds the `UserCodeNamespaceConfig` to the `Config`
+<4> Creates a `MapConfig` for the `map_name` Map, which references the specified namespace name
+<5> Puts the key and value in the Map
```

Comment:
```suggestion
<5> Puts the key and value in the Map, leveraging the namespace awareness
```
nit-pick

Reply:
what do you mean by "leveraging"? 

I know it is used extensively, but to me it is unclear. The definition is "using something that you already have to achieve something new or better". Could we say "using" or "to use", or something else simple and clear and straightforward.

If you feel leveraging is the best term, I'll go with it but only if I really have to; I recognise that this just a personal ugh word ;-)

Reply:
I'm sure @JamesHazelcast can come up with something better leveraging @rebekah-lawrence's feedback.

Reply:
My goal with this addition was to put more emphasis on the UCN aspect, as this particular example does not actually demonstrate typical UCN use-cases (putting a key & value into the map won't require any UCN resources besides maybe a `MapLoader` etc if it was configured).

I was using leveraging in the sense of:
"use (something) to maximum advantage.
'the organization needs to leverage its key resources'"

Perhaps my nit here would be better focused at the example - it should use something that implies user code was needed (like applying an `EntryProcessor`, which is usually our go-to demonstration). Again, happy to write a new example if that's of use to you!

Reply:
That would be perfect (again only if you have the time). Thank you

Reply:
Posted: https://github.com/hazelcast/hz-docs/pull/961#pullrequestreview-1860256896